pkg/cluster/fixssh.go

```diff
@@ -56,52 +74,99 @@ func (m *manager) fixSSH(ctx context.Context) error {
 		nicNameInstaller := fmt.Sprintf("%s-master%d-nic", infraID, i)
 		nicNameMachineAPI := fmt.Sprintf("%s-master-%d-nic", infraID, i)
 
-		var iErr, err error
 		var nic mgmtnetwork.Interface
 		nicName := nicNameInstaller
+		fallbackNIC := false
 
 		nic, err = m.interfaces.Get(ctx, resourceGroup, nicName, "")
 		if err != nil {
-			iErr = err // preserve original error
+			m.log.Warnf("Fetching details for NIC %s has failed with err %s", nicName, err)
+			fallbackNIC = true
+		} else if nic.InterfacePropertiesFormat != nil && nic.InterfacePropertiesFormat.VirtualMachine == nil {
+			err = m.removeBackendPoolsFromNIC(ctx, resourceGroup, nicName, &nic)
+			if err != nil {
+				m.log.Warnf("Removing BackendPools from NIC %s has failed with err %s", nicName, err)
+				return err
+			}
+			m.log.Warnf("Installer provisioned NIC %s has no VM attached", nicName)
+			fallbackNIC = true
+		}
+
+		if fallbackNIC {
 			nicName = nicNameMachineAPI
-			m.log.Warnf("fallback to check MachineAPI Nic name format for %s", nicName)
+			m.log.Warnf("Fallback to check MachineAPI Nic name format for %s", nicName)
 			nic, err = m.interfaces.Get(ctx, resourceGroup, nicName, "")
 			if err != nil {
-				m.log.Warnf("fallback failed with err %s", err)
-				return iErr
+				m.log.Warnf("Fallback failed with err %s", err)
+				return err
 			}
 		}
 
-		changed = updateNIC(&nic, &lb, i)
+		nicChanged := m.updateNIC(ctx, &nic, nicName, &lb, i, infraID)
 
-		if changed {
+		if nicChanged {
```

Comment:
It would also read better if we dropped the `changed` variable:
```go
if updateNIC(&nic, &lb, i) {
```

Reply:
Dropped `changed` variable in `updateNIC` function